neonvm/controllers/metrics.go

```diff
@@ -68,15 +68,47 @@ type wrappedReconciler struct {
 	failing map[client.ObjectKey]struct{}
 }
 
-// WithMetrics wraps a given Reconciler with metrics capabilities.
-func WithMetrics(reconciler reconcile.Reconciler, rm ReconcilerMetrics, cntrlName string) reconcile.Reconciler {
-	return &wrappedReconciler{
+// ReocncileSnapshot provides a glimpse into the current state of ongoing reconciles
+//
+// This type is (transitively) returned by the controller's "dump state" HTTP endpoint, and exists
+// to allow us to get deeper information on the metrics - we can't expose information for every
+// VirtualMachine into the metrics (it'd be too high cardinality), but we *can* make it available
+// when requested.
+type ReconcileSnapshot struct {
```

Comment:
It is a good idea to have `json` tags on a struct. Allows to rename fields, linters can catch when unexported fields have json tags, etc

I don't remember which particular problem I have faced, but I remember to always have json tags now :)

Reply:
Makes sense, yeah, will add that.

Reply:
done